core/http/src/main/java/org/trellisldp/http/impl/MutatingLdpHandler.java

```diff
@@ -98,11 +100,13 @@ protected MutatingLdpHandler(final TrellisRequest req, final ServiceBundler trel
      * @param entity the entity
      */
     protected MutatingLdpHandler(final TrellisRequest req, final ServiceBundler trellis,
-            final String baseUrl, final File entity) {
+            final String baseUrl, final InputStream entity) {
         super(req, trellis, baseUrl);
         this.entity = entity;
         this.session = ofNullable(req.getSecurityContext().getUserPrincipal()).map(Principal::getName)
             .map(getServices().getAgentService()::asAgent).map(HttpSession::new).orElseGet(HttpSession::new);
+        this.algorithms = trellis.getBinaryService().supportedAlgorithms().stream().map(String::toLowerCase)
```

Comment:
If we're gong to call this on every mutating request, we should memoize it and assume that the supported digest algorithms aren't going to change in between two requests.

Reply:
or, just call this when a digest-based request is used, which is already going to be a computationally expensive operation.